### PR TITLE
hourly recurrence

### PR DESCRIPTION
**Pull request by JozefCeluch**
I added hourly recurrence in recurrence dialog. I also disabled calling dismiss() on KEYCODE_BACK in RadialTimePickerDialog so that onCancel is called because onDismiss is called on each screen rotation and it is impossible to decide whether the user pressed back or the rotation changed

### CODE REVIEW

File: library/src/com/doomonafireball/betterpickers/radialtimepicker/RadialTimePickerDialog.java [hunk 1]
```diff
@@ -505,7 +505,7 @@ private void setCurrentItemShowing(int index, boolean animateCircle, boolean del
      * @return true if the key was successfully processed, false otherwise.
      */
     private boolean processKeyUp(int keyCode) {
-        if (keyCode == KeyEvent.KEYCODE_ESCAPE || keyCode == KeyEvent.KEYCODE_BACK) {
+        if (keyCode == KeyEvent.KEYCODE_ESCAPE ) {
```
Comment (fchauveau) [inline library/src/com/doomonafireball/betterpickers/radialtimepicker/RadialTimePickerDialog.java]: Why did you removed "KeyEvent.KEYCODE_BACK" from this check ?

**Comment by fchauveau** [comment]
Hi @dodkoC 
Its an interesting feature.

Can you rebase your PR from master ?
Thank you

**Comment by JozefCeluch** [comment]
Hi @fchauveau, done

File: library/src/main/java/com/codetroopers/betterpickers/radialtimepicker/RadialTimePickerDialog.java [hunk 2]
```diff
@@ -516,7 +516,7 @@ private void setCurrentItemShowing(int index, boolean animateCircle, boolean del
      * @return true if the key was successfully processed, false otherwise.
      */
     private boolean processKeyUp(int keyCode) {
-        if (keyCode == KeyEvent.KEYCODE_ESCAPE || keyCode == KeyEvent.KEYCODE_BACK) {
```
Comment (fchauveau) [inline library/src/main/java/com/codetroopers/betterpickers/radialtimepicker/RadialTimePickerDialog.java:519]: Why did you remove KEYCODE_BACK check ?

Comment (JozefCeluch) [inline library/src/main/java/com/codetroopers/betterpickers/radialtimepicker/RadialTimePickerDialog.java:519]: Otherwise there is no way to distinguish if the dialog was cancelled by user or dismissed (i.e. on screen rotation). By removing the check, onCancel() method is called for both tapping outside of the dialog and pressing the back button (i.e. the user decides to cancel the dialog). It allowed me to differentiate between the dialog being dismissed on rotation and cancelled by user.

Comment (fchauveau) [inline library/src/main/java/com/codetroopers/betterpickers/radialtimepicker/RadialTimePickerDialog.java:519]: Ok, 
So lets remove this from this PR and create a new one. So I could discuss with teammate what to do with this and merge to part about hour recurrence

**Comment by JozefCeluch** [comment]
@fchauveau OK, I removed the unrelated commit

**Comment by fchauveau** [comment]
merged, will be released soon
